### DISCUSSION TITLE
What is the difference between or-tools and python-mip, e.g. performance, correctness?

### DISCUSSION THREAD

**Discussion by Smartisa** [google/or-tools#3449] [Linear Solver questions] [answered]
or-tools claims the competition is consistently gold (https://developers.google.com/optimization) and python-mip claims to be the fastest open source linear programming solver (http://plato.asu.edu/ftp/lpsimp.html), so which one is faster?

EDIT: SO cross post: https://stackoverflow.com/questions/73552667

**Comment by Mizux** [comment, marked as answer]
Minizinc Challenge is a **constraint programing** competition
results: https://www.minizinc.org/challenge (note OR-Tools compete with the CP-SAT Solver)

python-mip seems to claims to be the best in a simplex **linear programming** competition
(note GLOP is part of OR-Tools as the Google Linear solver focused on numerical stability and precision)

also I can read:
> **Multi-solver**
Currently integrated with the C libraries of COIN-OR CBC solver and the commercial solver Gurobi. Other solvers will be supported soon.
src: https://www.python-mip.com/

So to me python-mip is just a wrapper on top of CBC (MIP solver) and Gurobi like we did with MPSolver in OR-Tools.
It could be interesting to compare both support using the same solver as backend to compare integration/config etc...

**Comment by lperron** [comment]
Please indicate when you cross post from stack overflow.
